birdyboard.py

```diff
@@ -0,0 +1,505 @@
+import pickle
+
+
+class Birdyboard:
+
+    def __init__(self):
+        self.users = []
+        self.user_name = None
+
+        self.chirps_library = {"public": [], "private": []}
+        # these variables are assigned when a specific chirp thread is being viewed or added to.
+        self.public_or_private = ""
+        self.chirp_index = None
+
+# ###############################
+# ######## MENU PRINTERS ########
+# ###############################
+
+        self.unlogged_in_menu = "Welcome. You are not logged in.\n1. new user\n2. log in\n3. view public chirps\n'x' to exit.\n"
+        self.logged_in_menu = "Welcome " + self.user_name + "!\n1. log out\n2. view public chirps\n3. view private chirps\n4. New public chirp thread\n5. new private chirp thread\n'x' to exit.\n"
```

Comment:
This throws a hard error and crashes out both your app and your tests, as setting the value of `self.logged_in_menu` requires converting `self.user_name` to a string, and at this point in the program flow, `self.user_name` is `None` and the NoneType can't be converted to a string.